base-template.sh

```diff
@@ -567,52 +689,81 @@ execute_shared_hooks() {
+            echo "  in the local or global Git configuration." >&2
+            echo ""
+            echo "  You need to fix this by running" >&2
+            echo "    \$ git hooks shared add [--local|--global] \"$SHARED_REPO\"" >&2
+            echo "  and deleting it from the \`.shared\` file by" >&2
+            echo "    \$ git hooks shared remove --shared \"$SHARED_REPO\"" >&2
+            return 1
         fi
 
-        # Note: GIT_DIR might be set (?bug?) (actually the case for post-checkout hook)
-        # which means we really need a `-f` to sepcify the actual config!
-        REMOTE_URL=$(git -C "$SHARED_ROOT" config -f "$SHARED_ROOT/.git/config" --get remote.origin.url)
-        ACTIVE_REPO=$(echo "$SHARED_REPOS_LIST" | grep -o "$REMOTE_URL")
-        if [ "$ACTIVE_REPO" != "$REMOTE_URL" ]; then
+        if [ ! -d "$SHARED_ROOT" ]; then
+
             echo "! Failed to execute shared hooks in $SHARED_REPO" >&2
-            echo "  The URL \`$REMOTE_URL\` is different." >&2
-            echo "  To fix it, run:" >&2
-            echo "    \$ git hooks shared purge" >&2
-            echo "    \$ git hooks shared update" >&2
+            if [ "$SHARED_REPO_IS_LOCAL" = "true" ]; then
+                echo "  It is not existing." >&2
+            else
+                echo "  It is not available. To fix, run:" >&2
+                echo "    \$ git hooks shared update" >&2
+            fi
 
-            if [ "$FAIL_ON_NOT_EXISTING" = "true" ]; then
+            if [ "$FAIL_ON_NOT_EXISTING_SHARED_HOOK" = "true" ]; then
                 return 1
             else
                 echo "  Continuing..." >&2
                 continue
             fi
         fi
 
+        if [ "$SHARED_REPO_IS_CLONED" = "true" ]; then
+            # Note: GIT_DIR might be set (?bug?) (actually the case for post-checkout hook)
+            # which means we really need a `-f` to sepcify the actual config!
+            REMOTE_URL=$(git -C "$SHARED_ROOT" config -f "$SHARED_ROOT/.git/config" --get remote.origin.url)
+            if ! echo "$SHARED_REPOS_LIST" | grep -q "$REMOTE_URL"; then
+                echo "! Failed to execute shared hooks in $SHARED_REPO" >&2
+                echo "  The remote URL \`$REMOTE_URL\` is different." >&2
+                echo "  To fix it, run:" >&2
+                echo "    \$ git hooks shared purge" >&2
+                echo "    \$ git hooks shared update" >&2
+
+                if [ "$FAIL_ON_NOT_EXISTING_SHARED_HOOK" = "true" ]; then
+                    return 1
+                else
+                    echo "  Continuing..." >&2
+                    continue
+                fi
+            fi
+        fi
+
         if [ -d "${SHARED_ROOT}/.githooks" ]; then
             execute_all_hooks_in "${SHARED_ROOT}/.githooks" "$@" || return 1
         elif [ -d "$SHARED_ROOT" ]; then
             execute_all_hooks_in "$SHARED_ROOT" "$@" || return 1
         fi
 
+        EXECUTED_SHARED_HOOKS="$SHARED_ROOT,$EXECUTED_SHARED_HOOKS"
```

Comment:
/nit new line might be a bit more robust than comma?
```suggestion
        EXECUTED_SHARED_HOOKS="$SHARED_ROOT
$EXECUTED_SHARED_HOOKS"
```